main.go

```diff
@@ -85,40 +89,54 @@ func checkEnvs(username string, token string) (string, string) {
 		}
 		token = tokenEnv
 	}
-	return username, token
+
+	pemEnv := os.Getenv("PEM_PATH")
+	if pemPath == "" {
+		if pemEnv == "" {
+			log.Fatal("path to pem file not set")
+		}
+		username = userEnv
+	}
+
+	pemPwdEnv := os.Getenv("PEM_PWD")
+	if pemPwd == "" {
+		if pemPwdEnv == "" {
+			log.Println("pem file password not set")
+		}
+		pemPwd = pemPwdEnv
+	}
+	return username, token, pemPath, pemPwd
 }
 
-func authType(repo string, username string, token string) transport.AuthMethod {
-	var auth transport.AuthMethod
+func authType(repo string, username string, token string, pemPath string, pemPwd string) transport.AuthMethod {
 	endpoint, err := transport.NewEndpoint(repo)
 	if err != nil {
 		log.Fatal(err)
 	}
 	switch {
 	case endpoint.Protocol() == "ssh":
-		pemPath := os.Getenv("PEM_PATH")
 		switch {
 		case pemPath != "":
 			user := os.Getenv("USER")
-			password := os.Getenv("PEM_TOKEN")
-			auth, err := ssh.NewPublicKeysFromFile(user, pemPath, password)
+			auth, err := ssh.NewPublicKeysFromFile(user, pemPath, pemPwd)
```

Comment:
What happens when Github username and local user differ?
What happens when username is not given?